src/cast_core/test/CrudActorYamlTests.yml

```diff
@@ -673,3 +673,30 @@ Tests:
     OutcomeData:
       - {a: 2}
     AssertNumTransactionsCommitted: 1
+
+  - Description: Test that an aggregate command with $out produces the expected data.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: "foo", b: 3 }
+            - WriteCommand: insertOne
+              Document: { a: "foo", b: 4 }
+      - OperationName: aggregate
+        OperationCommand:
+            Pipeline: [{$group: {_id: "$a", sum: {$sum: "$b"}}}, {$out: "test"}]
```

Comment:
Relying on `$out` here is maybe not ideal, but I couldn't figure out a better way to do it. For one thing, I didn't see any tests in this file for `OperationName: find`. The reason, I guess, is that genny throws away the results of the cursor, so there is no way to easily validate that the query is returning what we expect it to return. In the case of `$out`, the results of the query are persisted, so they can subsequently be validated using `OutcomeData`.

I considered testing this by trying to expose the metrics that are accounted for [here](https://github.com/mongodb/genny/pull/761/files#diff-d51310790eeb6ac904ed06a273fe737958403033c54d2a613a339a1cc61b8c3eR745-R746). I don't know the genny code too well, but it seemed to me like this wouldn't be very natural to implement. I'm also not sure if there are other genny self-tests that make assertions about metrics like these.

Reply:
This is actually pretty clever. Consider putting a version of your comment in the code, though, because the `$out` was easy (for me) to miss / understand why it was there without your verbiage.

It doesn't seem worthwhile to change much testing infra to support an alternative and I can't think of anything better off-hand, so my vote is to leave this and potentially reconsider if there's additional use-cases in the future.

Reply:
> This is actually pretty clever. Consider putting a version of your comment in the code, though, because the $out was easy (for me) to miss / understand why it was there without your verbiage.

Done. Let me know what you think of the comment.

> It doesn't seem worthwhile to change much testing infra to support an alternative and I can't think of anything better off-hand, so my vote is to leave this and potentially reconsider if there's additional use-cases in the future.

Sounds good! That's pretty much where I landed as well.